src/main/scala/zio/interop/reactivestreams/Adapters.scala

```diff
@@ -138,6 +138,9 @@ object Adapters {
       val subscriber =
         new Subscriber[A] {
 
+          def offerSafe(el: Exit[Option[Throwable], A]): UIO[Unit] =
+            ZIO.ifM(q.isShutdown)(ZIO.unit, q.offer(el).unit)
```

Comment:
How about replacing the `ifM` with an error handler? This way we do not have additional work on the happy path.

Scaladoc for `ZQueue.interrupt`: 
>  Interrupts any fibers that are suspended on `offer` or `take`. Future calls to `offer*` and `take*` will be interrupted immediately.

```suggestion
            q.offer(el).unit.catchAllCause(_ => UIO.unit)
```

Reply:
it looks like `.catchAllCause(_ => UIO.unit)` ignores all errors, `unsafeRunSync` does the same with even less work)

Reply:
That's even better. :+1: